### PR TITLE
Nn 3153 add the new select activity location page

### PR DESCRIPTION
**Pull request by JamieDraperUK**
image

### CODE REVIEW

File: integration-tests/integration/activity/activityList.spec.js [hunk 1]
```diff
@@ -231,24 +230,12 @@ context('Activity list page', () => {
     newSearchPage.period().should('have.value', initialPeriod)
   })
 
-  it('navigates to the whereabouts search on a page refresh', () => {
```
Comment (JamieDraperUK) [inline integration-tests/integration/activity/activityList.spec.js:234]: No longer relevant.

File: src/ResultsActivity/ResultsActivityContainer.js [hunk 2]
```diff
@@ -27,31 +30,24 @@ class ResultsActivityContainer extends Component {
...
     this.getActivityList = this.getActivityList.bind(this)
-    this.update = this.update.bind(this)
+    this.onListCriteriaChange = this.onListCriteriaChange.bind(this)
+    this.reload = this.reload.bind(this)
     this.state = {
       redactedPrint: false,
     }
   }
 
   async componentDidMount() {
```
Comment (JamieDraperUK) [inline src/ResultsActivity/ResultsActivityContainer.js:40]: This never really worked and resulted in the data that drives the page to be fetched 2-3 times on each state change.

File: src/ResultsActivity/ResultsActivityContainer.js [hunk 3]
```diff
@@ -27,31 +30,24 @@ class ResultsActivityContainer extends Component {
...
 
-  async componentDidUpdate(prevProps) {
-    const { date, period } = this.props
-
-    if ((prevProps.date && prevProps.date !== date) || (prevProps.period && prevProps.period !== period)) {
-      await this.update()
+    if (!date && !period && !location) {
+      window.location = '/manage-prisoner-whereabouts'
+    } else {
+      getAbsentReasonsDispatch()
```
Comment (JamieDraperUK) [inline src/ResultsActivity/ResultsActivityContainer.js:49]: Pulled the absent reason request up into componentDidMount so that it only gets called once.

**Review by jsrobertson — APPROVED** [review]
LGTM!

**Pull request merged**